src/dlgTriggerEditor.cpp

```diff
@@ -2406,69 +3274,70 @@ void dlgTriggerEditor::addTimer(bool isFolder)
+        mpVarsMainArea->lineEdit_var_name->setText(QString());
+        mpVarsMainArea->lineEdit_var_name->setPlaceholderText(tr("Table name..."));
 
         clearDocument(mpSourceEditorEdbee, QLatin1Literal("NewTable"));
-        name = "";
     } else {
         // Edbee doesn't have a readonly option, so I'm using setEnabled
         mpSourceEditorEdbee->setEnabled(true);
-        mpVarsMainArea->lineEdit_var_name->setText("");
-        mpVarsMainArea->lineEdit_var_name->setPlaceholderText("Variable name...");
-        mpVarsMainArea->var_type->setDisabled(false);
-        mpVarsMainArea->var_type->setCurrentIndex(0);
-        name = "";
+        mpVarsMainArea->lineEdit_var_name->setText(QString());
+        mpVarsMainArea->lineEdit_var_name->setPlaceholderText(tr("Variable name..."));
+        mpVarsMainArea->comboBox_variable_value_type->setDisabled(false);
+        mpVarsMainArea->comboBox_variable_value_type->setCurrentIndex(0);
     }
+
     QStringList nameL;
-    nameL << name;
+    nameL << QString();
     QTreeWidgetItem* cItem = treeWidget_variables->currentItem();
     LuaInterface* lI = mpHost->getLuaInterface();
     VarUnit* vu = lI->getVarUnit();
     TVar* cVar = vu->getWVar(cItem);
     QTreeWidgetItem* pParent;
-    QTreeWidgetItem* newItem;
+    QTreeWidgetItem* pNewItem;
     if (cVar && cVar->getValueType() == LUA_TTABLE) {
         pParent = cItem;
     } else {
         pParent = cItem->parent();
     }
+
     auto newVar = new TVar();
     if (pParent) {
         //we're nested under something, or going to be.  This HAS to be a table
         TVar* parent = vu->getWVar(pParent);
         if (parent && parent->getValueType() == LUA_TTABLE) {
             //create it under the parent
-            newItem = new QTreeWidgetItem(pParent, nameL);
+            pNewItem = new QTreeWidgetItem(pParent, nameL);
             newVar->setParent(parent);
         } else {
-            newItem = new QTreeWidgetItem(mpVarBaseItem, nameL);
+            pNewItem = new QTreeWidgetItem(mpVarBaseItem, nameL);
             newVar->setParent(vu->getBase());
         }
     } else {
-        newItem = new QTreeWidgetItem(mpVarBaseItem, nameL);
+        pNewItem = new QTreeWidgetItem(mpVarBaseItem, nameL);
         newVar->setParent(vu->getBase());
     }
+
     if (isFolder) {
-        newVar->setValue("", LUA_TTABLE);
+        newVar->setValue(QString(), LUA_TTABLE);
     } else {
         newVar->setValueType(LUA_TNONE);
     }
-    vu->addTempVar(newItem, newVar);
-    newItem->setFlags(newItem->flags() & ~(Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled));
-    if (newItem) {
-        mpCurrentVarItem = newItem;
-        treeWidget_variables->setCurrentItem(newItem);
+    vu->addTempVar(pNewItem, newVar);
+    pNewItem->setFlags(pNewItem->flags() & ~(Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled));
+// The following test is pointless - we will already have seg. faulted if pNewItem is a nullptr...!
```

Comment:
Have I got this right - `vu->addTempVar(...)` doesn't  modify its first argument to the extent that it might cease to exist, does it?